services/zone-service.ts

```diff
@@ -117,6 +122,82 @@ class ZoneService {
         return await zoneSchema.deleteOne({_id: zoneID});
     }
 
+    async editZone(zoneID: string | string[], newZone: InterestZone): Promise<InterestZone> {
+        let interestZone = await this.findById(zoneID);
+
+        await this.validateZoneEditting(newZone);
+
+        interestZone.address = {
+            ...interestZone.address,
+            name: newZone.address.name
+        };
+        interestZone.status = newZone.status;
+        interestZone.volunteerName = newZone.volunteerName;
+        interestZone.observations = newZone.observations;
+        interestZone.noUnsterilizedCats = newZone.noUnsterilizedCats;
+        interestZone.contactPerson = newZone.contactPerson;
+        
+        await interestZone.save();
+        return interestZone;
+    }
+
+    async deleteCat(zoneID: string | string[], catID: string | string[]): Promise<number> {
+        const interestZone = await this.findById(zoneID);
+        let cat = this.findCat(interestZone, catID, true);
+        if (!cat) {
+            cat = this.findCat(interestZone, catID, false);
+            if (cat) interestZone.noUnsterilizedCats--;
+        }
+        
+        if (!cat) throw new Error(`Cat with ${catID} not found`)
+
+        cat.remove();
+        interestZone.save();
+        return 1;
+    }
+
+    async editCat(zoneID: string | string[], catID: string | string[], newCat: any) : Promise<any> {
+        const interestZone = await this.findById(zoneID);
+        const sterilizedStatus: boolean = await this.validateCat(newCat);
+
+        let cat = this.findCat(interestZone, catID, sterilizedStatus);
+        if (!cat) throw new Error(ZoneError.CAT_NOT_FOUND);
+        
+        cat.gender = newCat.gender;
+        cat.mediaLinks = newCat.mediaLinks;
+        cat.observations = newCat.observations;
+
+        if (sterilizedStatus) {
+            cat.hospitalizationDate = newCat.hospitalizationDate;
+            cat.releaseDate = newCat.releaseDate;
+            cat.volunteerName = newCat.volunteerName;
+        }
+
+        await interestZone.save();
+        return cat;
+    }
+
+    async changeToSterilized(zoneID: string | string[], catID: string | string[], newCat: SterilizedCat): Promise<SterilizedCat> {
+        let interestZone = await this.findById(zoneID);
+
+        const cat = this.findCat(interestZone, catID, false);
+        if (!cat) throw new Error(`Cat with ${catID} not found`);
+
+        await this.addCatToZone(zoneID, {sterilizedStatus: true, ...newCat}, interestZone);
+        cat.remove();
+        interestZone.noUnsterilizedCats--;
+        await interestZone.save();
+        return interestZone;
```

Comment:
Please add an empty row in the block above, to group the lines of code into related blocks